code/common/strings.cc

```diff
@@ -27,6 +31,33 @@ namespace common::strings
 
     if (size < 1)
       return std::nullopt;
+    #elif defined(__linux__)
+    // Note: There is definitely some kind of template that will unify these
+    // conversions and make the code cleaner to look at/reduce code duplication
+
+    // Make a copy of the input string in case iconv alters it
+    // +1 to make room for a null-terminator we'll add on the end
+    size_t buf_in_size = wide.size() + 1;
+    wchar_t in_buf[buf_in_size];
+    std::wcsncpy(in_buf, wide.data(), wide.size());
+    // Ensure it's null-terminated: wstring_view doesn't guarantee null-termination
+    in_buf[wide.size()] = L'\0';
+
+    // Create pointers to be fed into iconv
+    char* ptr_in = reinterpret_cast<char*>(in_buf); // This is ok, iconv wants a wchar_t* casted to char*
+    size_t buf_out_size = buf_in_size*6; // UTF8 can be up to 6 bytes/symbol, wide is 1 element/symbol by def'n
+    char buf_out[buf_out_size];
+    char* ptr_out = static_cast<char*>(buf_out);
+
+    // wchar_t* to utf8 char*
+    // WCHAR_T should use whatever the system encoding for wchar_t* is
+    // //IGNORE flag will drop anything that can't be utf-8 encoded
+    auto descriptor = iconv_open("UTF-8//IGNORE", "WCHAR_T");
+    iconv(descriptor, &ptr_in, &buf_in_size, &ptr_out, &buf_out_size);
+    iconv_close(descriptor);
+
+    std::string out(buf_out);
+    #endif
```

Comment:
Not needed for non-windows

Reply:
Addressed in a1d45580444fd8a62b1644297758291c587b6198